app/models/course/forum/topic.rb

```diff
@@ -121,6 +121,11 @@ def update_resolve_status
     end
   end
 
+  def latest_history(limit: 5)
+    lastest_posts = posts.only_published_posts.reorder(created_at: :desc).limit(limit)
+    RagWise::DiscussionExtractionService.new(forum.course, self, lastest_posts).call
+  end
```

Comment:
I prefer this to be
```rb
  def get_recent_posts(limit: 5)
    posts.only_published_posts.reorder(created_at: :desc).limit(limit)
  end
```

Then you can apply the `RagWise::DiscussionExtractionService` elsewhere, keep the topic model free from RagWise concerns.